datalad/support/external_versions.py

```diff
@@ -150,6 +150,33 @@ def _get_system_7z_version():
     lgr.debug("Could not determine version of 7z from stdout. %s", out)
 
 
+def get_rsync_version():
+
+    # This does intentionally not query the version of rsync itself, but
+    # that of the debian package it's installed with. Reason is in gh-7320,
+    # which results in the need to detect a patched-by-ubuntu version of rsync
+    # and therefore the package version, not the result of `rsync --version`.
+    from datalad.utils import on_linux, get_linux_distribution
+    if on_linux:
+        dist = get_linux_distribution()[0]
+        if dist in ['debian', 'ubuntu']:
+            out = _runner.run(['apt-cache', 'policy', 'rsync'],
+                              protocol=StdOutErrCapture)
+            for line in out['stdout'].splitlines():
+                parts = line.split()
+                if parts[0] == 'Installed:':
+                    ver = LooseVersion(parts[1])
+                    break
+        # If we have a debian package version, use this as rsync version.
+        # Otherwise report what `rsync --version` itself has to say.
+        if ver:
+            return ver
```

Comment:
IMHO this (above) is overkill since we are just talking about either we should skip the test while testing against some older git-annex .

And since this functionality is already in `external_versions`, then why not to assign to be reported for `cmd:annex` and thus avoid manual `LooseVersion`'ing here, and then just `external_versions['cmd:rsync'] >=3.1.3 ` in the test?    Eventually that skip would be gone anyways whenever we boost minimal git-annex version beyond that one.

or am I still missing the point warranting above complexity?